sketch/src/core/sketch_support.ts

```diff
@@ -0,0 +1,327 @@
+/**
+ * Support modules used by SketchRNN
+ *
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Deals with decompressing b64 models to uint8 arrays.
+ *
+ * @param b64encoded String of b64 encoded data.
+ * @returns uint8 version of data.
+ */
+export function stringToUint8array(b64encoded: string) {
+  const u8: Uint8Array = new Uint8Array(atob(b64encoded)
+    .split("")
+    .map((c: string): number => {
+      return c.charCodeAt(0);
+    }));
+  return u8;
+}
+
+/**
+ * Converts b64 data into int16 array version.
+ *
+ * @param b64encoded String of b64 encoded data.
+ * @returns int16 version of data.
+ */
+export function stringToArray(b64encoded: string) {
+  const u: Uint8Array = stringToUint8array(b64encoded);
+  const result: Int16Array = new Int16Array(u.buffer);
+  return result;
+}
+
+// Functions for sampling normal or bi-normals
+
+// Random numbers util (from https://github.com/karpathy/recurrentjs)
+let returnV = false;
+let vVal = 0.0;
+export function gaussRandom(): number {
+  if (returnV) {
+    returnV = false;
+    return vVal;
+  }
+  const u = 2 * Math.random() - 1;
+  const v = 2 * Math.random() - 1;
+  const r = u * u + v * v;
+  if (r === 0 || r > 1) {
+    return gaussRandom();
+  }
+  const c = Math.sqrt(-2 * Math.log(r) / r);
+  vVal = v * c; // cache this
+  returnV = true;
+  return u * c;
+}
+export function randf(a: number, b: number): number {
+  return Math.random() * (b - a) + a;
+}
+export function randi(a: number, b: number): number {
+  return Math.floor(Math.random() * (b - a) + a);
+}
+export function randn(mu: number, std: number): number {
+  return mu + gaussRandom() * std;
+}
+// From http://www.math.grin.edu/~mooret/courses/math336/bivariate-normal.html
+export function birandn(
+  mu1: number,
+  mu2: number,
+  std1: number,
+  std2: number,
+  rho: number): number[] {
+  const z1 = randn(0, 1);
+  const z2 = randn(0, 1);
+  const x = Math.sqrt(1 - rho * rho) * std1 * z1 + rho * std1 * z2 + mu1;
+  const y = std2 * z2 + mu2;
+  return [x, y];
+}
+
+/**
+ * Sample from a categorial distribution
```

Comment:
Add params, etc. to public functions.

Reply:
Not sure I know what you mean. These are only used by model.ts (and public functions have been added for the line simplification function to userline, via model.ts, for instance simplifyLine).

Reply:
In that case, you should hide these functions by removing the import from `index.ts`. `model.ts` can still import via relative paths.